fpmsyncd/routesync.cpp

```diff
@@ -752,88 +798,178 @@ void RouteSync::onRouteMsg(int nlmsg_type, struct nl_object *obj, char *vrf)
     string gw_list;
     string intf_list;
     string mpls_list;
-    getNextHopList(route_obj, gw_list, mpls_list, intf_list);
-    string weights = getNextHopWt(route_obj);
+    string nhg_id_key;
 
-    vector<string> alsv = tokenize(intf_list, NHG_DELIMITER);
-    for (auto alias : alsv)
+#ifdef HAVE_NEXTHOP_GROUP
+    uint32_t old_nhg_id = 0;
+    bool delete_route = false;
+    uint32_t nhg_id = rtnl_route_get_nh_id(route_obj);
+    if(nhg_id)
     {
-        /*
-         * An FRR behavior change from 7.2 to 7.5 makes FRR update default route to eth0 in interface
-         * up/down events. Skipping routes to eth0 or docker0 to avoid such behavior
-         */
-        if (alias == "eth0" || alias == "docker0")
+        bool use_nhg = false;
+        const auto itg = m_nh_groups.find(nhg_id);
+        if(itg == m_nh_groups.end())
+        {
+            SWSS_LOG_DEBUG("NextHop group id %d not found. Dropping the route %s", nhg_id, destipprefix);
+            return;
+        }
+        NextHopGroup& nhg = itg->second;
+        if(hasIntfNextHop(nhg))
         {
-            SWSS_LOG_DEBUG("Skip routes to eth0 or docker0: %s %s %s",
-                    destipprefix, gw_list.c_str(), intf_list.c_str());
-            // If intf_list has only this interface, that means all of the next hops of this route 
-            // have been removed and the next hop on the eth0/docker0 has become the only next hop. 
-            // In this case since we do not want the route with next hop on eth0/docker0, we return. 
-            // But still we need to clear the route from the APPL_DB. Otherwise the APPL_DB and data 
-            // path will be left with stale route entry
-            if(alsv.size() == 1)
+            string nexthops, ifnames, weights;
+
+            getNextHopGroupFields(nhg, nexthops, ifnames, weights, rtnl_route_get_family(route_obj));
+            FieldValueTuple gw("nexthop", nexthops.c_str());
+            FieldValueTuple intf("ifname", ifnames.c_str());
+            fvVector.push_back(gw);
+            fvVector.push_back(intf);
+            if(!weights.empty())
             {
-                if (!warmRestartInProgress)
-                {
-                    SWSS_LOG_NOTICE("RouteTable del msg for route with only one nh on eth0/docker0: %s %s %s %s",
-                            destipprefix, gw_list.c_str(), intf_list.c_str(), mpls_list.c_str());
+                FieldValueTuple wg("weight", weights.c_str());
+                fvVector.push_back(wg);
+            }
+            use_nhg = false;
+            SWSS_LOG_DEBUG("NextHop group id %d has interface nexthop address. Filling the route table %s with nexthop and ifname", nhg_id, destipprefix);
+        }
+        else
+        {
+            nhg_id_key = getNextHopGroupKeyAsString(nhg_id);
+            FieldValueTuple nhg("nexthop_group", nhg_id_key.c_str());
+            fvVector.push_back(nhg);
+            updateNextHopGroup(nhg_id);
+            use_nhg = false;
```

Comment:
use_nhg=true?

Reply:
This code was removed after the change to not use NHG for route with single nexthop.